internal/kafka/internal/services/kafka.go

```diff
@@ -502,6 +505,22 @@ func (k *kafkaService) PrepareKafkaRequest(kafkaRequest *dbapi.KafkaRequest) *er
 		return errors.NewWithCause(errors.ErrorGeneral, err, "error managing certificate for kafka %q", kafkaRequest.ID)
 	}
 
+	if k.kafkaTLSCertificateManagementService.IsAutomaticCertificateManagementEnabled() {
+		_, err = k.kafkaTLSCertificateManagementService.GetCertificate(context.Background(), kafkatlscertmgmt.GetCertificateRequest{
+			TLSCertRef: kafkaRequest.KafkasRoutesBaseDomainTLSCrtRef,
+			TLSKeyRef:  kafkaRequest.KafkasRoutesBaseDomainTLSKeyRef,
+		})
+
+		if err != nil {
+			if !coreErrors.Is(err, fs.ErrNotExist) {
+				return errors.NewWithCause(errors.ErrorGeneral, err, "error checking if certificate for kafka %q has been created", kafkaRequest.ID)
+			}
+
+			// do not continue the reconciliation of a preparing Kafka if the certificate has not been generated yet.
+			return nil
+		}
+	}
```

Comment:
If we are managing cert automatically, we wait for the cert to be generated before we can move the state of the Kafka from preparing to provisioning